Numerous Balance Changes & Additions

nekomaster1000 opened the pull request:
- Cured Jerky's Speed II effect now lasts for 15 seconds.
- Raw Hogchops now poison you with 10 seconds of Infection II, while Cooked Hogchops give you 10 seconds of Slowness II (to balance out their incredible hunger restoration)
- Added the Spirit Eye, which gives you 10 seconds of Glowing when you eat it.
- Changed the Tongue Whip to deal 6 points of damage instead of 3, in addition to making it charge faster by 7 ticks.
- Added the Kinetic Tongue Whip, a Mining Master compatibility weapon. Deals 8 points of damage.
- Removed bizarro 'Dullthorn_block' texture.
- Updated Dullthorns Block texture palette to be consistent with the current Dullthorns textures.
- Updated the Dullthorns Tip texture so that if the tiling breaks and repeats the top texture, it should still look somewhat good.
- Glowsilk now repairs 2000 Durability at the cost of 5 levels, from a previous 200 Durability at 4 levels cost.
- Updated Glowsilk Bow & Ascus Bomb sprites.
- Reduced Blindsights' Luminance bestowing to 5 seconds from 30.
- Blindsights now give you 10 seconds of Jump Boost II on-hit.
- Blindsight Tongue is now consumable for 5 seconds of Jump Boost II.
- Deleted the Skeletal Piglin Loot table :(
- Blackstone Dwarves now only drop an Apple if killed by the Player.
- Blindsights, Shroomloins and Hoglins drop their loot less often.
- Made Shroomloins far harder to anger, removing most blocks from their anger goals.
- It should no longer rain in the Glowstone Canyon (and Delta Shores.)
- Fixed Infection Overlay displaying incorrectly.

To Do:
Kinetic Tongue Whip needs to utilises Knockback III in addition to being given more durability.

Thank you Ludo for helping catch my mistakes and implementing the Hogchop drop-rate change and Kinetic Tongue Whip rendering!

nekomaster1000 committed:
Numerous Balance Changes & Additions

- Cured Jerky's Speed II effect now lasts for 15 seconds.
- Raw Hogchops now poison you with 10 seconds of Infection II, while Cooked Hogchops give you 10 seconds of Slowness II (to balance out their incredible hunger restoration)
- Added the Spirit Eye, which gives you 10 seconds of Glowing when you eat it.
- Changed the Tongue Whip to deal 6 points of damage instead of 3, in addition to making it charge faster by 7 ticks.
- Added the Kinetic Tongue Whip, a Mining Master compatibility weapon. Deals 8 points of damage.
- Removed bizarro 'Dullthorn_block' texture.
- Updated Dullthorns Block texture palette to be consistent with the current Dullthorns textures.
- Updated the Dullthorns Tip texture so that if the tiling breaks and repeats the top texture, it should still look somewhat good.
- Glowsilk now repairs 2000 Durability at the cost of 5 levels, from a previous 200 Durability at 4 levels cost.
- Updated Glowsilk Bow & Ascus Bomb sprites.
- Reduced Blindsights' Luminance bestowing to 5 seconds from 30.
- Blindsights now give you 10 seconds of Jump Boost II on-hit.
- Blindsight Tongue is now consumable for 5 seconds of Jump Boost II.
- Deleted the Skeletal Piglin Loot table :(
- Blackstone Dwarves now only drop an Apple if killed by the Player.
- Blindsights, Shroomloins and Hoglins drop their loot less often.
- Made Shroomloins far harder to anger, removing most blocks from their anger goals.
- It should no longer rain in the Glowstone Canyon (and Delta Shores.)
- Fixed Infection Overlay displaying incorrectly.

To Do:
Kinetic Tongue Whip needs to utilises Knockback III in addition to being given more durability.

Thank you Ludo for helping catch my mistakes and implementing the Hogchop drop-rate change and Kinetic Tongue Whip rendering!

SwanX1 commented:
Should blindsight attacks give you a positive effect?

SwanX1 commented on src/main/java/org/infernalstudios/infernalexp/init/IEItems.java:
```diff
@@ -16,28 +16,18 @@
-import org.infernalstudios.infernalexp.items.WhipItem;
-import org.infernalstudios.infernalexp.items.IESpawnEggItem;
+import org.infernalstudios.infernalexp.blocks.VerticalSlabBlock;
+import org.infernalstudios.infernalexp.items.*;
```
\>:(

nekomaster1000 commented:
The Tongue Stew and Tongue both give you Jump Boost, so it was added to the Blindsight's attack too. You could argue it's somewhat negative when the Blindsight applies it to you as it could momentarily mess with your spacial awareness if you're jumping to fight it.

SwanX1 commented on src/main/java/org/infernalstudios/infernalexp/init/IEItems.java:
```diff
@@ -115,7 +121,8 @@
     // Tools
     public static final RegistryObject<BowItem> GLOWSILK_BOW = registerItem("glowsilk_bow", () -> new GlowsilkBowItem(new Item.Properties().maxDamage(384).group(InfernalExpansion.TAB)));
-    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 1.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
+    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 4.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
```
I thought the damage was balanced, why the change? I mean, you can literally enchant it with sharpness

nekomaster1000 replied:
Nobody wanted to use the Whip with it's previous damage output. It's new damage is akin to an Iron Sword, which I feel is appropriate.

SwanX1 commented on src/main/java/org/infernalstudios/infernalexp/init/IEItems.java:
```diff
@@ -136,4 +143,12 @@ public static void register(IEventBus eventBus) {
         return ITEMS.register(name, itemSupplier);
     }
 
+
+    public static <T extends Item> RegistryObject<T> registerItemConditioned(String name, Supplier<? extends T> itemSupplier, Supplier<? extends T> itemSupplierCompat, String modID) {
+            RegistryObject<T> item = ITEMS.register(name,ModList.get().isLoaded(modID)? itemSupplier:itemSupplierCompat);
+
+            return item;
+    }
+
+
```
tooo many empty lines

SwanX1 replied:
Do you really need another method for this?
See my code suggestion above

SwanX1 replied:
~~Skill~~ Formatting issue:
```suggestion
    public static <T extends Item> RegistryObject<T> registerItemConditioned(String name, Supplier<? extends T> itemSupplier, Supplier<? extends T> itemSupplierCompat, String modID) {
        RegistryObject<T> item = ITEMS.register(name, ModList.get().isLoaded(modID) ? itemSupplier : itemSupplierCompat);
        return item;
    }
```

SwanX1 commented on src/main/java/org/infernalstudios/infernalexp/init/IELootModifiers.java:
```diff
@@ -54,6 +54,11 @@ protected HoglinLootModifier(ILootCondition[] conditionsIn) {
         @Nonnull
         @Override
         protected List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context) {
+
+            if ((context.getRandom().nextDouble() < 0.6)) {
+                return generatedLoot;
+            }
+
             if (!InfernalExpansionConfig.MobInteractions.USE_HOGCHOPS.getBoolean()) {
                 return generatedLoot;
             }
```
```suggestion
            if (context.getRandom().nextDouble() < 0.6 || !InfernalExpansionConfig.MobInteractions.USE_HOGCHOPS.getBoolean()) {
                return generatedLoot;
            }
```

SwanX1 replied:
```suggestion
            if (context.getRandom().nextDouble() < 0.6 || !InfernalExpansionConfig.MobInteractions.USE_HOGCHOPS.getBoolean()) {
                return generatedLoot;
            }
```

SwanX1 commented on src/main/java/org/infernalstudios/infernalexp/init/IEItems.java:
```diff
@@ -115,7 +121,8 @@
     // Tools
     public static final RegistryObject<BowItem> GLOWSILK_BOW = registerItem("glowsilk_bow", () -> new GlowsilkBowItem(new Item.Properties().maxDamage(384).group(InfernalExpansion.TAB)));
-    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 1.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
+    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 4.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
```
Why change the attack damage? I think it was balanced before

SwanX1 commented on src/main/java/org/infernalstudios/infernalexp/init/IEItems.java:
```diff
@@ -115,7 +121,8 @@
     public static final RegistryObject<BowItem> GLOWSILK_BOW = registerItem("glowsilk_bow", () -> new GlowsilkBowItem(new Item.Properties().maxDamage(384).group(InfernalExpansion.TAB)));
-    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 1.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
+    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 4.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
+    public static final RegistryObject<WhipItem> KINETIC_TONGUE_WHIP = registerItemConditioned("kinetic_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 6.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)), () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 6.0F, -3.4F, new Item.Properties()),"miningmaster");
```
```suggestion
    public static final RegistryObject<WhipItem> KINETIC_TONGUE_WHIP = registerItem("kinetic_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 6.0F, -3.4F, ModList.get().isLoaded("miningmaster") ? new Item.Properties().group(InfernalExpansion.TAB) : new Item.Properties()));
```

SwanX1 commented on src/main/resources/assets/infernalexp/models/item/spirit_eye.json:
```diff
@@ -1,6 +1,6 @@
   "parent": "item/generated",
   "textures": {
-    "layer0": "infernalexp:item/cured_jerky"
+    "layer0": "infernalexp:item/spirit_eye"
```
lmao what

SwanX1 commented:
LGTMn't

nekomaster1000 committed:
Update src/main/java/org/infernalstudios/infernalexp/init/IELootModifiers.java

From Swan.

Co-authored-by: Kārlis Čerņavskis <karliscern@gmail.com>

nekomaster1000 committed:
Update src/main/java/org/infernalstudios/infernalexp/init/IEItems.java

From Swan.

Co-authored-by: Kārlis Čerņavskis <karliscern@gmail.com>

nekomaster1000 committed:
Update src/main/java/org/infernalstudios/infernalexp/init/IEItems.java

From Swan.

Co-authored-by: Kārlis Čerņavskis <karliscern@gmail.com>

nekomaster1000 committed:
Glowsilk Bow Sprite Fix + Balance Reversions

- Thanks to Hellion for talking me out of the Glowsilk balance change! Keeping it's extra level cost though.
- Also, reverted the config-breaking Jerky buff. Will

nekomaster1000 committed:
Fixed Imports and Whip Item changes

Dariensg reviewed (approved):
LGTM

Dariensg committed:
Merge branch '1.16' into masterchanges1.16-2.5

Dariensg committed:
Removed unnecessary method
